### PR TITLE
Update dependencies for hello world apps

### PR DESCRIPTION
**Pull request by rahulPalliyalil**
This PR updates the dependencies for the hello world services dp-hello-world-api, dp-hello-world-controller and dp-hello-world-event

### CODE REVIEW

**Review by justinpjose — COMMENTED** [review]
There are some changes in the project-generation templates in `dp-cli` which hasn't been reflected in this PR. For example;

The current README is in `dp-hello-world-event` is out of date - https://github.com/ONSdigital/dp-hello-world/blob/master/dp-hello-world-event/README.md. The README should be updated as reflected in dp-cli - https://github.com/ONSdigital/dp-cli/blob/main/project_generation/content/templates/event/README.md.tmpl

I think this happened because you updated this repo manually. I would suggest to run the project-generation using the dp-toolkit (information given in the dp-cli README) for each dummy dp-hello-world service so that you get all the updates

**Comment by rahulPalliyalil** [comment]
@justinpjose the reason why i had to do this manually was, there were lots of changes which was not merged into the hello world as this has moved away quite a lot. So would you like me to go head with the generation then ?

**Comment by nshumoogum** [comment]
@rahulPalliyalil - as Justin suggested use the dp-cli generation tool to update the 3 apps in dp-hello-world, if there are changes due to changes in the generation tool that you did not make. I would explain that to the developer reviewing when they leave a comment.

**Comment by rahulPalliyalil** [comment]
> @rahulPalliyalil - as Justin suggested use the dp-cli generation tool to update the 3 apps in dp-hello-world, if there are changes due to changes in the generation tool that you did not make. I would explain that to the developer reviewing when they leave a comment and say they are welcome to add a change in dp-cli generation tool and regenerate this code from it and have a new review.
> 
> But the fact of the matter is that the dp-cli generation tool has been signed off and repos are using this code whether it is in here or not.
> 
> The process for changing the dp-cli generation tool:
> 
> 1. Create new branch on dp-cli and make changes to templates in app generation tool
> 2. Regenerate apps or individual apps in dp-hello-world so developers can review the changes and comment (via a PR)
> 3. Once approved in dp-hello-world, merge code in dp-hello-world
> 4. Create PR in dp-cli, changes in PR may not only be template changes but script changes too, so we need sign off on those too
> 5. Once approved in dp-cli, merge and start using generation tool to generate new apps

@nshumoogum  and @justinpjose  I have now regenerated all the hello world apps from the dp cli. If you can please have a look, that would be god.

File: dp-hello-world-api/ci/unit.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ image_resource:
   type: docker-image
   source:
     repository: golang
-    tag: 1.15.1
+    tag: 1.16.7
```
Comment (nshumoogum) [inline dp-hello-world-api/ci/unit.yml:9]: Should this be 1.17.7?

File: dp-hello-world-event/go.mod [hunk 2]
```diff
@@ -3,25 +3,14 @@ module github.com/ONSdigital/dp-hello-world-event
 go 1.15
```
Comment (nshumoogum) [inline dp-hello-world-event/go.mod:3]: Should this be 1.17

**Review by nshumoogum — CHANGES_REQUESTED** [review]
Just make sure the generation tool also updates the go version in go.mod and ci files to 1.17.7

**Comment by justinpjose** [comment]
> There are some changes in the project-generation templates in `dp-cli` which hasn't been reflected in this PR. For example;
> 
> The current README is in `dp-hello-world-event` is out of date - https://github.com/ONSdigital/dp-hello-world/blob/master/dp-hello-world-event/README.md. The README should be updated as reflected in dp-cli - https://github.com/ONSdigital/dp-cli/blob/main/project_generation/content/templates/event/README.md.tmpl
> 
> I think this happened because you updated this repo manually. I would suggest to run the project-generation using the dp-toolkit (information given in the dp-cli README) for each dummy dp-hello-world service so that you get all the updates

This example which I gave is still not in the changes. Are you sure that you re-generated for event driven apps?

File: dp-hello-world-api/ci/unit.yml (hunk 1 above)
Comment (rahulPalliyalil) [reply dp-hello-world-api/ci/unit.yml]: Fixed in commit 2.

File: dp-hello-world-event/go.mod (hunk 2 above)
Comment (rahulPalliyalil) [reply dp-hello-world-event/go.mod]: Fixed in commit 2 .

**Review by nshumoogum — APPROVED** [review]

**Review by justinpjose — APPROVED** [review]

**Pull request merged**